packages/block-library/src/navigation-link/edit.js

```diff
@@ -333,6 +333,7 @@ export default function NavigationLinkEdit( {
 			isKeyboardEvent.primary( event, 'k' ) ||
 			( ( ! url || isDraft || isInvalid ) && event.keyCode === ENTER )
 		) {
+			event.preventDefault();
```

Comment:
It would be cool to add some inline comments to explain why we needed the prevent default.

Reply:
See https://github.com/WordPress/gutenberg/pull/59864